### PR TITLE
Added commercial building analysis

### PR DESCRIPTION
**Pull request by Vismayak**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,13 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+
```
Comment (longshuicy) [inline CHANGELOG.md:8]: nitpicking.. extra empty line 8 and line 14

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+        Returns:
+            pd.DataFrame: Total delay time of all impeding factors from REDi framework.
+        """
+
+        # Obtain the commercial buildings in damage
+        damage = mcs_failure[mcs_failure['haz_expose'] == 'yes']
+        commercial = []
+        for i, b in enumerate(buildings):
+            if b['properties']['archetype'] in (6, 7, 8, 15, 16, 18, 19):
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:136]: the archetypes are hardcoded here. is there a better way? Maybe pass in that as a parameter?

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+        recovery = self.recovery_rate(buildings, sample_damage_states, total_delay)
+        end_recovery = time.process_time()
+        print("Finished executing recovery_rate() in " + str(end_recovery - end_total_delay) + " secs")
+
+        time_stepping_recovery = CommercialBuildingRecovery.time_stepping_recovery(recovery)
+        end_time_stepping_recovery = time.process_time()
+        print("Finished executing time_stepping_recovery() in " +
+              str(end_time_stepping_recovery - end_recovery) + " secs")
+
+        #result = time_stepping_recovery
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:109]: is this commented out line needed?

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+
+            # Now, perform the two nested loops, using the indexing function to simplify the syntax.
+            for i in range(0, num_samples):
+                state = samples_mcs_ds[i]
+                #lognormal_mean = mapped_repair.repair_curves[state].alpha
+                #lognormal_sdv = mapped_repair.repair_curves[state].beta
+
+                # Generate a vector of random values to use in the inner loop
+                #rand_vals = np.random.random(num_samples)
+                #repair_time = lognorm.ppf(rand_vals, lognormal_sdv, scale=np.exp(lognormal_mean)) / 7
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:269]: Same for those inline comments. Are they needed?

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+
+        Args:
+            recovery_results (pd.DataFrame): Total recovery time of financial delay and other factors from REDi framework.
+
+        Returns:
+            pd.DataFrame: Time formatted recovery results.
+
+        """
+        time_step = 90 / 7  # a quarter in weeks
+        year = 4
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:301]: can those be parameters with default?

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+        # Convert recovery time results to numpy
+        samples_n1_n2 = recovery_results.drop(columns=['guid']).to_numpy()
+
+        # Number of guids
+        num_buildings = recovery_results.shape[0]
+        num_samples = samples_n1_n2.shape[1]
+        num_times = len(total_time)
+
+        # Generate a numpy to hold the results as desired
+        times_np = np.full((num_buildings, num_times), 1111.0)
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:317]: What's 1111.0 in this case?

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+                },
+                {
+                    'id': 'recovery',
+                    'description': 'CSV file of commercial building recovery time',
+                    'type': 'incore:buildingRecovery'
+                },
+                {
+                    'id': 'time_stepping_recovery',
+                    'description': 'CSV file of commercial building recovery percent',
+                    'type': 'incore:buildingRecovery'
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:415]: All three output datasets have the same type. Is it intended?

**Review by longshuicy — COMMENTED** [review]
Don't forget to include those .rst file for incore docs

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 8]
```diff
@@ -0,0 +1,418 @@
...
+                                                delay_vec[contractor_idx]
+                                            ]) +
+                                            delay_vec[permit_idx])
+
+        total_delay = pd.DataFrame(samples_np, columns=colnames)
+        total_delay.insert(0, 'guid', merged_delay_guids)
+
+        return total_delay
+
+    def recovery_rate(self, buildings, sample_damage_states, total_delay):
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:205]: At first glance, this looks identical to the function in `pyincore/analyses/residentialbuildingrecovery/residentialbuildingrecovery.py.` You might be able to just reuse that

File: pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+
+        # Now, generate all the labels using list comprehension outside the loops
+        colnames = [f'sample_{i}_{j}' for i in range(0, num_samples) for j in range(0, num_samples)]
+        recovery_time = pd.DataFrame(samples_n1_n2, columns=colnames)
+        recovery_time.insert(0, 'guid', merged_delay_guids)
+
+        return recovery_time
+
+    @staticmethod
+    def time_stepping_recovery(recovery_results):
```
Comment (longshuicy) [inline pyincore/analyses/commercialbuildingrecovery/commercialbuildingrecovery.py:290]: This also looks identical to the one in residentialbuildingrecovery

**Review by longshuicy — COMMENTED** [review]
Could you update the module.rst to include this new analysis? 
Also if possible, could you create a corresponding task for creating markdowns and jupyter notebook in incore-docs? Thanks!

**Review by longshuicy — APPROVED** [review]

**Review by ylyangtw — APPROVED** [review]
changes look good

**Pull request merged**